index.bs

```diff
@@ -1444,19 +1441,46 @@ Because of that, JavaScript authors take for granted
 that the data available to a function won’t change unexpectedly
 while the function is running.
 
-So if a JavaScript Web API exposes some piece of data,
-such as an object property,
-the user agent must not update that data
-while a JavaScript task is running.
-Instead, if the underlying data changes,
-<a>queue a task</a> to modify the exposed version of the data.
+Changes that are not the result of developer action
+and changes that are asynchronously delivered
+should not happen in the middle of other JavaScript,
+including between [=microtasks=].
+Instead, a task should be [[html#queuing-tasks|queued]],
+or they should update as part of [=update the rendering=].
+
+<div class="example">
+During a `while (true)`,
+and after `await`ing an already-resolved `Promise`,
+you *shouldn't* expect things like:
+
+* The DOM to update as a result of the HTML parser loading network content
+* {{HTMLImageElement/width|img.width}} to change as a result of loading image data from the network
+* Buttons of a {{Gamepad}} to change state
+* {{Element/scrollTop}} to change, even if scrolling can visually occur
+
+These things aren't updated by the currently running script,
+so they shouldn't change during the current task.
+
+</div>
+
+Data can update synchronously from the result of developer action.
 
 <div class="example">
-If a JavaScript task has accessed the {{NavigatorOnline/onLine|navigator.onLine}} property,
-and browser's online status changes,
-the property won't be updated until the next task runs.
+{{ChildNode/remove()|node.remove()}} changes the DOM synchronously
+and is immediately observable.
 </div>
 
+A few kinds of situations justify violating this rule:
+
+* Observing the current time,
+    as in {{Date/now|Date.now()}} and {{Performance/now|performance.now()}},
+    although note that it's also useful to present a consistent task-wide time
+    as in {{AnimationTimeline/currentTime|document.timeline.currentTime}}.
+* Functions meant to help developers interrupt synchronous work,
+    as in the case of the proposed {{isInputPending()}}.
```

Comment:
This feature is somewhat controversial because it violates run-to-completion.

Reply:
Feels reasonable to drop this reference. The other examples are enough.

Reply:
The TAG is on record supporting this exception in https://github.com/w3ctag/design-reviews/issues/415#issuecomment-527655955 and https://github.com/w3ctag/design-reviews/issues/475#issuecomment-621438616 (cc @dbaron for any historical context). I can break this into a separate PR if we need to discuss it more, but this function seems to be a case of putting authors ahead of technical purity. There might also be a better phrasing to explain what makes this exception acceptable?